src/app/context.tsx

```diff
@@ -0,0 +1,27 @@
+'use client';
+
+import React, { ReactNode, createContext, useContext, useState } from 'react';
+import { NewsWithId } from './models/types';
+
+type NewsState = {
+  context: NewsWithId[] | null;
```

Comment:
1. Rename it to news
2. Arrays are quite convenient for types. Please remove `| null` and simply use `NewsWithId[]`